submissions/AriadnaKravchuk/opp-tiny-js-world/index.js

```diff
@@ -0,0 +1,125 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/AriadnaKravchuk/a-tiny-JS-world
+   Web app: https://ariadnakravchuk.github.io/a-tiny-JS-world
+*/
+
+// ======== OBJECTS DEFINITIONS ========
+
+const SPECIES = {
+   HUMAN: "human",
+   CAT: "cat",
+   DOG: "dog"
+}
+
+const GENDER = {
+   MALE: "male",
+   FEMALE: "female"
+}
+
+const SAYING = {
+   DOG: "woof-woof!",
+   CAT: "meow!"
+}
+
+class Inhabitant {
+   constructor(species, name, gender, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+      this.friends = [];
+      this.attributes = [];
+   }
+
+   addFriends(...friend) {
+      for (let i = 0; i < friends.length; i++) {
+         if (!this.friends.includes(friends[i])) {
+            this.friends.push(friends[i]);
+         }
+      }
+   }
+
+   printAttributes() {
+      let friendsNames = [];
+
+      for (let i = 0; i < this.friends.length; i++) {
+         friendsNames.push(this.friends[i]["name"]);
+      }
+
+      print(this.attributes.join("; ") + "; " + friendsNames.join(", "));
```

Comment:
(1) Make class agnostic of environment it works in.
Let the method just return whatever it needs to create and other code to use it (e.g. print).

(2) Separation of concerns. Currently this method does two things: form an output string and prints it.